vernac/declare.ml

```diff
@@ -1881,6 +1870,53 @@ let prepare_proof ?(warn_incomplete=true) { proof } =
   in
   proofs, Evd.evar_universe_context evd
 
+exception NotGuarded of
+    Environ.env * Evd.evar_map *
+    (Environ.env * int * EConstr.t Type_errors.pcofix_guard_error) option *
+    (Environ.env * int * int list * EConstr.t Type_errors.pfix_guard_error) list *
```

Comment:
Why are we keeping copies of the env?

Reply:
The first env  is the context for the whole declaration. The other envs are the contexts for the respective errors.

Reply:
are they not the same?